### DISCUSSION TITLE
Tutorial is not working

### DISCUSSION THREAD

**Discussion by dineshsiprtc** [remix-run/react-router#11151] [Q&A]
I started the tutorial as explained in the tutorial. I stacked on the page when introduce the confirmation box. where is the confirmation box.Please provide the tutorial full source code on the github so that we can learn.

Thanks

**Comment by kiliman** [comment]
Can you link to the section in the tutorial you're having problems with?

As for the final result, try this:

https://github.com/remix-run/examples/tree/main/_official-contacts-tutorial

**Reply by dineshsiprtc** [reply]
I may attach screenshot of the page.

https://reactrouter.com/en/main/start/tutorial

![image](https://github.com/remix-run/remix/assets/150108336/13d01290-3315-42b3-a6fd-83ef95223349)

I wanted to run this tutorial for my learning

**Reply by sergiodxa** [reply]
I just moved this discussion to the RR repo since you linked to the RR docs.

Is your issue with the confirm function? That’s a browser function.

**Reply by dineshsiprtc** [reply]
in react using plain confirm give error in react. we have to use as window.confirm.

src/routes/contact.jsx don't have getContacts   ... Please guide

![image](https://github.com/remix-run/react-router/assets/150108336/7f76dfe9-9719-45c7-a7a9-da6fe784c9e5)
